lib/improver/utilities/cube_metadata.py

```diff
@@ -652,45 +653,49 @@ def add_history_attribute(cube, value, append=False):
         cube.attributes["history"] = new_history
 
 
+def probability_cube_name_regex(cube_name):
+    """
+    Regular expression matching IMPROVER probability cube name.  Returns
+    None if the cube_name does not match the regular expression (ie does
+    not start with 'probability_of').
+
+    Args:
+        cube_name (str):
+            Probability cube name
+    """
+    regex = re.compile(
+        '(probability_of_)'  # always starts this way
+        '(?P<diag>.*?)'      # named group for the diagnostic name
+        '(_in_vicinity|)'    # optional group, may be empty
+        '(?P<thresh>_above_threshold|_below_threshold|_between_thresholds|$)')
```

Comment:
^ The regex string could be a global variable eg. PROB_CUBE_REGEX out top

Reply:
It could - I quite like having it close to the functions where it's used, though, so the developer can see what's going on.  But I'll leave this for the second reviewer to consider.

Thanks for reviewing this.